src/components/Profile/ProfileForm.js

```diff
@@ -0,0 +1,145 @@
+import {
+  FormGroup,
+  TextInput,
+  InlineNotification,
+  Form,
+} from "carbon-components-react";
+import React from 'react';
+import styled from "styled-components";
+
+import { spacing, colors } from "../../theme";
+import { ResponsiveContainer } from '../pageStyles';
+import Button from "../Button";
+import AddressForm from "./AddressForm";
+
+const Row = styled.div`
+  display: grid;
+  grid-auto-columns: 1fr;
+  grid-column-gap: ${spacing[5]};
+  grid-template-columns: 1fr 1fr;
+
+  div {
+    grid-column-end: unset;
+  }
+`;
+
+const Divider = styled.div`
+  height: 1px;
+  width: 100%;
+  background-color: ${colors.gray[20]};
+  margin-bottom: ${spacing[5]};
+`;
+
+const InlineNotificationStyled = styled(InlineNotification)`
+  width: 100%;
+  max-width: 100%;
+  margin-bottom: ${spacing[3]};
+`;
+
+const SubmissionContainer = styled.div`
+  margin-top: ${spacing[8]};
+`;
+
+export const ProfileForm = ({ ambassador, setAmbassador, err, disablePhone, disableEmail }) => <ResponsiveContainer>
+  <Form onSubmit={async (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+
+    const userData = {
+      first_name: formData.get("first_name"),
+      last_name: formData.get("last_name"),
+      email: formData.get("email"),
+      phone: String(formData.get("phone") || ""),
+      date_of_birth: formData.get("date_of_birth"),
+      address: {
+        address1: formData.get("address1"),
+        state: formData.get("state"),
+        zip: formData.get("zip"),
+        city: formData.get("city"),
+      },
+      form_submitted: true,
+    };
+
+    await setAmbassador((data) => {
+      return {
+        ...data,
+        ...userData,
+      };
+    });
+  }}>
+    <FormGroup legendText="">
+      <Row>
+        <TextInput
+          id="first_name"
+          name="first_name"
+          invalidText="Invalid error message."
+          labelText="First Name*"
+          defaultValue={ambassador.first_name}
+          required
+        />
+        <TextInput
+          id="last_name"
+          name="last_name"
+          invalidText="Invalid error message."
+          labelText="Last Name*"
+          defaultValue={ambassador.last_name}
+          required
+        />
+      </Row>
+    </FormGroup>
+    <FormGroup legendText="">
+      <TextInput
+        id="phone"
+        name="phone"
+        invalidText="Invalid error message."
+        labelText="Phone number*"
+        defaultValue={ambassador.phone}
+        required
+        disabled={disablePhone}
+      />
+    </FormGroup>
+    <FormGroup legendText="">
+      <TextInput
+        id="email"
+        name="email"
+        invalidText="Invalid error message."
+        labelText="Email"
```

Comment:
@zestyping, could you change the label to "Email*" and pass the disableEmail prop through to this component? These changes were a part of making email not editable by the user but they seem to have got lost. I put them back in the original file on another branch I've PR'ed so that will have conflicts, also.

Reply:
@shaboptimal Okay I fixed this!  I see what happened—it wasn't lost in this change, but in my previous change to make the form prefillable (that PR was opened a few days ago).  I've restored the change; apologies!